docs/development.md

```diff
@@ -10,30 +12,52 @@ Run `ng generate component component-name` to generate a new component. You can
 
 NOTES: To generate code outside the `src/app` directory,  you must either be in that directory or provide a path relative to the `src/app` directory:
 
-```
-cd $PROJECT_ROOT
-ng generate pipe ../app-shared/core/pipes/example-pipe
-# OR
-cd $PROJECT_ROOT/src/app-shared
-ng generate pipe ./core/pipes/example-pipe
-```
+    cd $PROJECT_ROOT
+    ng generate pipe ../app-shared/core/pipes/example-pipe
+    # OR
+    cd $PROJECT_ROOT/src/app-shared
+    ng generate pipe ./core/pipes/example-pipe
 
 
 ## Build
 
-Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory. Use the `--prod` flag for a production build.
+Run `npm run build:prod` to build the project. The build artifacts will be stored in the `dist/` directory.
 
 
 ## Running unit tests
 
-Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.io).
+Run `npm run test` to execute the unit tests via [Karma](https://karma-runner.github.io).
 
-To run unit tests in a headless browser use: `ng test --configuration=ci`
+To run unit tests in a headless browser use: `npm run test:ci`
 
 
 ## Running end-to-end tests
 
-Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).
+Before running the tests, create an `environments/e2e.env` configuration file based on `environments/sample-e2e.env`.
+
+Run `npm run e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).
+
+To run unit tests in a headless browser use: `npm run e2e:ci`
+
+To run or debug e2e tests in IDEs, you first need to run `npm run e2e` in order to prepare the WebDriver files.
+
+[protractor.angular.io][https://protractor.angular.io/] contains information on how to write e2e tests. 
+
+You can also use the protractor CLI directly in order to avoid having to wait for Angular to rebuild while only working on tests:
+
+    # Run a development server in one terminal:
+    npm run start
+
+    # Install dependencies for running e2e tests
+    npx webdriver-manager update
+
+    # Run e2e tests in another terminal
+    npx protractor ./e2e/protractor-ci.conf.js
+
+    # You can also limit the number of tests tests are being executed:
+    npx protractor ./e2e/protractor-ci.conf.js --specs ./e2e/src/account/login.e2e-spec.ts
+    # or use RegExp to only run certain tests, like tests that end with 'invalid credentials$':
+    npx protractor ./e2e/protractor-ci.conf.js --grep='invalid credentials$'
```

Comment:
During development, it's best to just start the Angular CLI development server in one terminal and then manually run the tests (without using the Angular CLI) in another tab (or inside IDEs). If you do this, the existing 12 e2e tests finish in about 14s, when using the Angular CLI command, they finish in 30s, because the Angular CLI also builds the application and checks if the tests dependencies are present.